java/arcs/core/analysis/ClaimDeduction.kt

```diff
@@ -0,0 +1,150 @@
+    }
+
+    override fun visit(expr: Expression.TextLiteralExpression): Paths {
+        TODO("Not yet implemented")
+    }
+
+    override fun visit(expr: Expression.BooleanLiteralExpression): Paths {
+        TODO("Not yet implemented")
+    }
+
+    override fun visit(expr: Expression.NullLiteralExpression): Paths {
+        TODO("Not yet implemented")
+    }
+
+    override fun visit(expr: Expression.FromExpression): Paths {
+        TODO("Not yet implemented")
+    }
+
+    override fun visit(expr: Expression.WhereExpression): Paths {
+        TODO("Not yet implemented")
+    }
+
+    override fun <T> visit(expr: Expression.SelectExpression<T>): Paths {
+        TODO("Not yet implemented")
+    }
+
+    override fun visit(expr: Expression.LetExpression): Paths {
+        TODO("Not yet implemented")
+    }
+
+    override fun <T> visit(expr: Expression.FunctionExpression<T>): Paths {
+        TODO("Not yet implemented")
+    }
+}
+
+/** A visitor that accumulates [ClaimDerivations] from a Paxel [Expression]. */
+class ExpressionClaimDeducer : Expression.Visitor<ClaimDerivations> {
+
+    override fun <E, T> visit(expr: Expression.UnaryExpression<E, T>): ClaimDerivations {
+        TODO("Not yet implemented")
+    }
+
+    override fun <L, R, T> visit(expr: Expression.BinaryExpression<L, R, T>): ClaimDerivations {
+        return expr.left.accept(this) + expr.right.accept(this)
+    }
+
+    override fun <T> visit(expr: Expression.FieldExpression<T>): ClaimDerivations {
+        if (expr.qualifier == null) {
+            return emptyMap()
+        }
+        return expr.qualifier!!.accept(this)
+    }
+
+    override fun <T> visit(expr: Expression.QueryParameterExpression<T>): ClaimDerivations {
+        TODO("Not yet implemented")
+    }
+
+    override fun visit(expr: Expression.NumberLiteralExpression): ClaimDerivations {
+        TODO("Not yet implemented")
+    }
+
+    override fun visit(expr: Expression.TextLiteralExpression): ClaimDerivations {
+        TODO("Not yet implemented")
+    }
+
+    override fun visit(expr: Expression.BooleanLiteralExpression): ClaimDerivations {
+        TODO("Not yet implemented")
+    }
+
+    override fun visit(expr: Expression.FromExpression): ClaimDerivations {
+        TODO("Not yet implemented")
+    }
+
+    override fun visit(expr: Expression.WhereExpression): ClaimDerivations {
+        TODO("Not yet implemented")
+    }
+
+    override fun <T> visit(expr: Expression.SelectExpression<T>): ClaimDerivations {
+        TODO("Not yet implemented")
+    }
+
+    override fun visit(expr: Expression.LetExpression): ClaimDerivations {
+        TODO("Not yet implemented")
+    }
+
+    override fun <T> visit(expr: Expression.FunctionExpression<T>): ClaimDerivations {
+        TODO("Not yet implemented")
+    }
+
+    override fun visit(expr: Expression.NewExpression) =
+        expr.fields.associateBy(
+            { it.first.split(".").map { AccessPath.Selector.Field(it) }.toList() },
```

Comment:
`toList()` is redundant? Also, passing named arguments makes it more readable.
```suggestion
            keySelector = { (fieldName, _) -> 
               fieldName.split(".").map { AccessPath.Selector.Field(it) } 
            },
```